ovn-tester/ovn_workload.py

```diff
@@ -567,91 +596,102 @@ def __init__(self, central_node, worker_nodes, cluster_cfg, brex_cfg):
+                                                        self.cluster_cfg.vips)
+        await self.load_balancer.add_vips(self.cluster_cfg.static_vips)
 
-    def create_cluster_join_switch(self, sw_name):
-        self.join_switch = self.nbctl.ls_add(sw_name,
-                                             self.cluster_cfg.gw_net)
+    async def create_cluster_join_switch(self, sw_name):
+        self.join_switch = await self.nbctl.ls_add(sw_name,
+                                                   self.cluster_cfg.gw_net)
 
         lrp_ip = netaddr.IPAddress(self.cluster_cfg.gw_net.last - 1)
-        self.join_rp = self.nbctl.lr_port_add(
+        self.join_rp = await self.nbctl.lr_port_add(
             self.router, 'rtr-to-join', RandMac(), lrp_ip,
             self.cluster_cfg.gw_net.prefixlen
         )
-        self.join_ls_rp = self.nbctl.ls_port_add(
+        self.join_ls_rp = await self.nbctl.ls_port_add(
             self.join_switch, 'join-to-rtr', self.join_rp
         )
 
-    def provision_ports(self, n_ports, passive=False):
-        return [
-            self.select_worker_for_port().provision_ports(self, 1, passive)[0]
-            for _ in range(n_ports)
-        ]
-
-    def unprovision_ports(self, ports):
+    async def provision_ports(self, n_ports, passive=False):
+        tasks = []
+        ctx = ovn_context.active_context
+        for _ in range(n_ports):
+            worker = self.select_worker_for_port()
+            tasks.append(ctx.create_task(worker.provision_ports(self, 1,
+                                                                passive)))
+        results = await asyncio.gather(*tasks)
+        return [ports[0] for ports in results]
+
+    async def unprovision_ports(self, ports):
         for port in ports:
             worker = port.metadata
-            worker.unprovision_port(self, port)
+            await worker.unprovision_port(self, port)
 
-    def ping_ports(self, ports):
+    async def ping_ports(self, ports):
         ports_per_worker = defaultdict(list)
         for p in ports:
             ports_per_worker[p.metadata].append(p)
         for w, ports in ports_per_worker.items():
-            w.ping_ports(self, ports)
+            await w.ping_ports(self, ports)
 
     @ovn_stats.timeit
-    def provision_vips_to_load_balancers(self, backend_lists):
-        n_vips = len(self.load_balancer.vips.keys())
-        vip_ip = self.cluster_cfg.vip_subnet.ip.__add__(n_vips + 1)
-
-        vips = {
-            f'{vip_ip + i}:{DEFAULT_VIP_PORT}':
-                [f'{p.ip}:{DEFAULT_BACKEND_PORT}' for p in ports]
-            for i, ports in enumerate(backend_lists)
-        }
-        self.load_balancer.add_vips(vips)
-
-    def unprovision_vips(self):
-        self.load_balancer.clear_vips()
-        self.load_balancer.add_vips(self.cluster_cfg.static_vips)
+    async def provision_vips_to_load_balancers(self, backend_lists):
+        async with self.lock:
```

Comment:
Same comment as for `Namespace.provision_vips_to_load_balancers()`, I think we can avoid the lock.